core/src/main/java/hudson/widgets/BuildHistoryWidget.java

```diff
@@ -52,17 +52,17 @@ public BuildHistoryWidget(Task owner, Iterable<T> baseList, Adapter<? super T> a
     /**
      * Returns the first queue item if the owner is scheduled for execution in the queue.
      */
-    public Item getQueuedItem() {
+    public QueueItem getQueuedItem() {
```

Comment:
Since we are incompatibly changing the signature, it would be a good time to restrict `BuildHistoryWidget`. (After verifying from https://github.com/jenkins-infra/usage-in-plugins that it is in fact not used from plugins. Unfortunately the default report https://ci.jenkins.io/job/Infra/job/usage-in-plugins/job/master/lastSuccessfulBuild/artifact/output/ shows only usages of _deprecated_ members so it is useless for this purpose.)

Reply:
https://github.com/search?q=org%3Ajenkinsci+BuildHistoryWidget&type=code shows usage in
* batch-task-plugin -- https://github.com/jenkinsci/batch-task-plugin/blob/1b92e1749646186e73e076f9a9b41e48ba6f8455/src/main/java/hudson/plugins/batch_task/BatchTask.java#L20
* DotCI -- https://github.com/jenkinsci/DotCi/blob/611dae322da72898ed75bcf635da89cb24b1cbd7/src/main/java/com/groupon/jenkins/dynamic/build/DynamicSubProject.java#L48. This extends `Project` so the method should be deleted.

Reply:
The BuildHistoryWidget feature is broken in the `batch-task-plugin` and I wish to get it to work again. [I haven't been able to use the Job/Run implementations](https://issues.jenkins.io/browse/JENKINS-33833), and wonder if I can get it to work with the QueueItem one. Is there any guideline on how to use @Beta features?

Reply:
> how to use beta features?

```xml
<useBeta>true</useBeta>
```

in your POM.